### PR TITLE
Fixes #1717 - ignore FutureWarnings which are raised by pandas

### LINKED ISSUE #1717: Cython module with import triggers deep introspection for pandas, raises unhandled FutureWarning
This is a somewhat complicated situation to reproduce, but basically `pandas` throws `FutureWarning`s for certain attributes, and when you import it into a Cython module (triggering astroid's deep module inspection), these future warnings are not handled by astroid and bubble up as `AstroidError`s through to pylint. Here is a full repro:

### Cython module `pyx.pyx`

```python
# distutils: language = c++
import pandas as pd

cdef class Test:
    def __cinit__(self):
        ...
```

### Python module `test.py`

```python
import pyx

pyx.Test()
```

### Commands
```
cythonize -a -i pyx.pyx
pylint --extension-pkg-allow-list=pyx,pandas test.py
```

### Exception
```
Exception on node <Import l.1 at 0x106b23ca0> in file '/Users/timkpaine/Programs/projects/other/astroid/test.py'
Traceback (most recent call last):
  File "/opt/homebrew/lib/python3.9/site-packages/pylint/checkers/imports.py", line 765, in _get_imported_module
    return importnode.do_import_module(modname)
  File "/opt/homebrew/lib/python3.9/site-packages/astroid/mixins.py", line 102, in do_import_module
    return mymodule.import_module(
  File "/opt/homebrew/lib/python3.9/site-packages/astroid/nodes/scoped_nodes/scoped_nodes.py", line 527, in import_module
    return AstroidManager().ast_from_module_name(absmodname)
  File "/opt/homebrew/lib/python3.9/site-packages/astroid/manager.py", line 168, in ast_from_module_name
    return self.ast_from_module(module, modname)
  File "/opt/homebrew/lib/python3.9/site-packages/astroid/manager.py", line 265, in ast_from_module
    return AstroidBuilder(self).module_build(module, modname)
  File "/opt/homebrew/lib/python3.9/site-packages/astroid/builder.py", line 91, in module_build
    node = self.inspect_build(module, modname=modname, path=path)
  File "/opt/homebrew/lib/python3.9/site-packages/astroid/raw_building.py", line 311, in inspect_build
    self.object_build(node, module)
  File "/opt/homebrew/lib/python3.9/site-packages/astroid/raw_building.py", line 367, in object_build
    self.object_build(module, member)
  File "/opt/homebrew/lib/python3.9/site-packages/astroid/raw_building.py", line 325, in object_build
    member = getattr(obj, name)
  File "/opt/homebrew/lib/python3.9/site-packages/pandas/__init__.py", line 198, in __getattr__
    warnings.warn(
FutureWarning: pandas.Float64Index is deprecated and will be removed from pandas in a future version. Use pandas.Index with the appropriate dtype instead.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/opt/homebrew/lib/python3.9/site-packages/pylint/utils/ast_walker.py", line 90, in walk
    callback(astroid)
  File "/opt/homebrew/lib/python3.9/site-packages/pylint/checkers/imports.py", line 472, in visit_import
    imported_module = self._get_imported_module(node, name)
  File "/opt/homebrew/lib/python3.9/site-packages/pylint/checkers/imports.py", line 788, in _get_imported_module
    raise astroid.AstroidError from e
astroid.exceptions.AstroidError
************* Module test
test.py:1:0: F0002: test.py: Fatal error while checking 'test.py'. Please open an issue in our bug tracker so we address this. There is a pre-filled template that you can use in '/Users/timkpaine/Library/Caches/pylint/pylint-crash-2022-07-19-17.txt'. (astroid-error)
```

### Standalone (Non Cython) repro for convenience

```python
import types
import pandas as pd
from astroid.builder import AstroidBuilder


m = types.ModuleType("test")
m.pd = pd

AstroidBuilder().module_build(m, "test")
```

xref: https://github.com/PyCQA/pylint/issues/7205
xref: https://github.com/PyCQA/astroid/pull/1719

### PR DESCRIPTION
**Pull request by timkpaine**
Long description and repro in https://github.com/PyCQA/astroid/issues/1717

## Steps

- [x] For new features or bug fixes, add a ChangeLog entry describing what your PR does.
- [x] Write a good description on what the PR does.

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |

## Related Issue

https://github.com/PyCQA/astroid/issues/1717
https://github.com/PyCQA/pylint/issues/7205

### CODE REVIEW

**Review by jacobtylerwalls — COMMENTED** [review]
Thank you for submitting a patch!

I looked at the blame to find why we're even escalating warnings to errors in the first place. It looks like the author was somewhat uncertain and it escaped discussion during review, see #785. I would be in favor of just changing the warningsfilter to "ignore" instead of "error" and then removing the warning types from the `except`. Otherwise, we could be back here with `UserWarning` or anything else. What do you think?

File: astroid/raw_building.py [hunk 1]
```diff
@@ -390,7 +390,7 @@ def object_build(
                 with warnings.catch_warnings():
                     warnings.simplefilter("error")
                     member = getattr(obj, name)
-            except (AttributeError, DeprecationWarning):
+            except (AttributeError, DeprecationWarning, FutureWarning):
                 # damned ExtensionClass.Base, I know you're there !
                 attach_dummy_node(node, name)
                 continue
```
Comment (Pierre-Sassoulas) [inline astroid/raw_building.py:396]: I agree with @jacobtylerwalls that we should (try to) change the filtering logic to not escalate to error (it's risky if we miss one escalation, on the top of my head we're missing UserWarning, PendingDeprecationWarning and SyntaxWarning at least).Maybe something like this ?
```python
with warnings.catch_warnings():
    warnings.simplefilter("ignore")
    member = getattr(obj, name)
```    

 I'm wondering what we do in  ``attach_dummy_node(node, name)`` that could necessitate to list everything. But I just checked and we indeed missed a lot of warning, it's actually surprising that this was not reported before : https://docs.python.org/3/library/warnings.html#warning-categories. Are those specific warnings never used in any library pylint astroid is parsing ?

**Review by Pierre-Sassoulas — COMMENTED** [review]
Thank you for opening the issue and fixing the problem yourself, much appreciated !

**Comment by timkpaine** [comment]
Another option is to just catch the base warning class, but I agree just ignoring it all is probably better and certainly cleaner.

**Review by Pierre-Sassoulas — APPROVED** [review]
👌

**Review by jacobtylerwalls — CHANGES_REQUESTED** [review]
Glad to have the fix in. I checked the regression test, and it [passes](https://github.com/jacobtylerwalls/astroid/pull/72) without your changes. Maybe a test that just uses `unittest.mock` to emit a `FutureWarning` as a side effect would be simpler?

This is a minor detail, but if you do end up needing to create a test module, perhaps you can put it with the others under `tests/testdata/python3/data`.

**Comment by timkpaine** [comment]
@jacobtylerwalls I accidentally broke the test by adding `Float64Index=""` to the fake module, if you remove that it should fail. I've fixed that test and moved the module inside the data folder

**Review by jacobtylerwalls — APPROVED** [review]

**Pull request merged**